mediorum/server/repair.go

```diff
@@ -81,6 +81,8 @@ func (ss *MediorumServer) startRepairer() {
 			tracker.AbortedReason = "NOT_ENOUGH_PEERS"
 			tracker.FinishedAt = time.Now()
 			saveTracker()
+			// wait 1 minute before running again
+			time.Sleep(time.Minute * 1)
```

Comment:
we already sleep at the beginning of this function to give it time to find healthy peers. if it needs more time then the sleep at the top of the func should be increased instead of decreasing here. also, i don't think running repair more frequently would decrease CPU, so i'm a little confused by that part of the PR description

Reply:
that sleep is outside of the loop isn't it?

Reply:
yeah it waits 1 minute to start the loop because it's giving peer_health.go some time to find healthy peers. this condition here shouldn't be reached (i've never seen it reached in stage or prod). and if it is then you'll see AbortedReason=NOT_ENOUGH_PEERS at the `/internal/logs/repair` route

Reply:
i was seeing like 30-40 of those logs per second and 100% cpu use on dev before this change... maybe it is something else?

Reply:
i've seen similar log spam when running with audius-compose before but forget what exactly it said. i would be surprised if anything in repair.go is the cause. lemme try locally and see if the logs give any info

Reply:
i saw the unable to find peers and this PR fixed it. lmk! i'll try again as well in a bit...

Reply:
ohhh i totally misread this! you're adding an additional sleep. yeah no harm there, sorry for the confusion. i think it never has enough healthy peers locally because it wants 3 peers, but there are only 3 total nodes. so it would be good to change the check to `len(healthyPeers) + 1 < ss.Config.ReplicationFactor` (add the `+ 1`)

Reply:
i see gotcha. can do that sep. i think this is still the right behavior. a full disk would also nuke the thing basically and crush our logs

Reply:
true yeah it can probably even be like a full 10 minute sleep for disk/peers. confirmed locally btw that it was this log and that R=3 and peers=2, so that len(healthyPeers)+1 fix should work

`2023-10-10 21:52:22 {"time":"2023-10-11T04:52:22.937257551Z","level":"WARN","source":{"function":"mediorum/server.(*MediorumServer).startRepairer","file":"/app/server/repair.go","line":78},"msg":"not enough healthy peers to run repair","self":"http://audius-protocol-creator-node-2","task":"repair","run":1,"cleanupMode":true,"R":3,"peers":2}`

anyway looks good to merge! great improvement

Reply:
🫡